pkg/cli/cves_sub_cmd.go

```diff
@@ -0,0 +1,205 @@
+		RunE: func(cmd *cobra.Command, args []string) error {
+			searchConfig, err := GetSearchConfigFromFlags(cmd, searchService)
+			if err != nil {
+				return err
+			}
+
+			err = CheckExtEndPointQuery(searchConfig, ImageListForCVEQuery())
+			if err != nil {
+				return fmt.Errorf("%w: '%s'", err, ImageListForCVEQuery().Name)
+			}
+
+			searchedCVEID := args[0]
+			ctx, cancel := context.WithCancel(context.Background())
+			retryCount := 1
+
+			return RetryWithContext(ctx, func() error {
+				err := SearchImagesByCVEIDGQL(searchConfig, repo, searchedCVEID)
+				if err != nil {
+					if !strings.Contains(err.Error(), zerr.ErrCVEDBNotFound.Error()) {
+						cancel()
+
+						return err
+					}
+
+					fmt.Fprintln(searchConfig.resultWriter,
+						"[warning] CVE DB is not ready [", retryCount, "] - retry in ", cveDBRetryInterval, " seconds")
+
+					retryCount += 1
+				}
+
+				return err
+			}, maxRetries, cveDBRetryInterval*time.Second)
+		},
+	}
+
+	imagesByCVEIDCmd.Flags().StringVar(&repo, "repo", "", "Search for a specific CVE by name/id")
+
+	return imagesByCVEIDCmd
+}
+
+func NewFixedTagsCommand(searchService SearchService) *cobra.Command {
+	fixedTagsCmd := &cobra.Command{
+		Use:   "fixed [repo] [cveId]",
+		Short: "List tags which have fixed a CVE",
+		Long:  `List tags which have fixed a CVE`,
+		Args: func(cmd *cobra.Command, args []string) error {
+			const argCount = 2
+
+			if err := cobra.ExactArgs(argCount)(cmd, args); err != nil {
+				return err
+			}
+
+			if !zcommon.CheckIsRepoName(args[0]) {
+				return fmt.Errorf("%w: expected a valid repo name for first argument '%s'", zerr.ErrInvalidCLIParameter, args[0])
+			}
+
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			searchConfig, err := GetSearchConfigFromFlags(cmd, searchService)
+			if err != nil {
+				return err
+			}
+
+			err = CheckExtEndPointQuery(searchConfig, ImageListWithCVEFixedQuery())
+			if err != nil {
+				return fmt.Errorf("%w: '%s'", err, ImageListWithCVEFixedQuery().Name)
+			}
+
+			repo := args[0]
+			searchedCVEID := args[1]
+			ctx, cancel := context.WithCancel(context.Background())
+			retryCount := 1
+
+			return RetryWithContext(ctx, func() error {
+				err := SearchFixedTagsGQL(searchConfig, repo, searchedCVEID)
+				if err != nil {
+					if !strings.Contains(err.Error(), zerr.ErrCVEDBNotFound.Error()) {
+						cancel()
+
+						return err
+					}
+
+					fmt.Fprintln(searchConfig.resultWriter,
+						"[warning] CVE DB is not ready [", retryCount, "] - retry in ", cveDBRetryInterval, " seconds")
+
+					retryCount += 1
+				}
+
+				return err
+			}, maxRetries, cveDBRetryInterval*time.Second)
+		},
+	}
+
+	return fixedTagsCmd
+}
+
+func RetryWithContext(ctx context.Context, operation func() error, maxRetries int,
+	delay time.Duration,
+) error {
+	err := operation()
+
+	for attempt := 0; err != nil && attempt < maxRetries; attempt++ {
+		timer := time.NewTimer(delay)
+
+		select {
+		case <-timer.C:
+			break
```

Comment:
Instead of creating a new timer in each iteration its probably best if we use a Ticker instead for this use case. Look at: https://gobyexample.com/tickers

Reply:
In case we'll want to add a exponential backoff delay I think we should use the `time.After(delay)` function. It's conveying the idea of a delay a bit better in this context, a ticker makes me think of a small operation repeated every tick but our operation can take several seconds.

Reply:
Ack.